File: VehicleInsuranceClaimApp/app/src/main/java/com/example/vehicleinsuranceclaimapp/database/AppDatabase.java
package com.example.vehicleinsuranceclaimapp.database;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.DatabaseConfiguration;
import androidx.room.InvalidationTracker;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteOpenHelper;

import com.example.vehicleinsuranceclaimapp.dao.ClaimDao;
import com.example.vehicleinsuranceclaimapp.dao.ClaimImageDao;
import com.example.vehicleinsuranceclaimapp.dao.PolicyDao;
import com.example.vehicleinsuranceclaimapp.dao.UserDao;
import com.example.vehicleinsuranceclaimapp.model.Claim;
import com.example.vehicleinsuranceclaimapp.model.ClaimImage;
import com.example.vehicleinsuranceclaimapp.model.Policy;
import com.example.vehicleinsuranceclaimapp.model.User;

@Database(entities = {Claim.class, User.class, Policy.class, ClaimImage.class}, version = 6 )
public abstract class AppDatabase extends RoomDatabase {

      public abstract ClaimDao claimDao();
      public abstract UserDao userDao();
      public abstract PolicyDao policyDao();
      public abstract ClaimImageDao claimImageDao();
      private  static AppDatabase instance;

      public static synchronized AppDatabase getInstance(Context context){
          if(instance == null){
              instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "claim_database").fallbackToDestructiveMigration().build();
          }
          return instance;
      }

}

File: FinalMilestone/app/src/main/java/com/example/finalmilestone/database/AppDatabase.kt
package com.example.finalmilestone.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.finalmilestone.dao.RecipeDao
import com.example.finalmilestone.model.Recipe
import com.example.finalmilestone.model.RecipeCategory

@Database(entities = [RecipeCategory::class, Recipe::class], version = 1)
abstract class AppDatabase: RoomDatabase() {
    abstract fun recipeDao(): RecipeDao

    companion object{
        @Volatile
        private var  INSTANCE: AppDatabase? = null

        fun getDatabase(context: Context) : AppDatabase{
            return INSTANCE ?: synchronized(this){
                val instance = Room.databaseBuilder(context.applicationContext,
                    AppDatabase::class.java,
                    "recipe_app_db"
                    ).build()
                INSTANCE = instance
                instance
            }
        }
    }
}